pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -517,7 +565,10 @@ func (nim *nodeInfoManager) tryUninstallDriverFromCSINode(
 			hasModified = true
 		}
 	}
-	if !hasModified {
+
+	annotationModified := setMigrationAnnotation(nim.migratedPlugins, nodeInfo)
```

Comment:
why would we modify the annotation on driver uninstall?

Reply:
you're right, its not strictly necessary. Figured I might as well check for modification on all events, this actually made me think about "what if the csinodeinfo already exists on Kubelet startup" so I have modified code to support that case as well.